src/app/pages/dashboard/dashboard.component.ts

```diff
@@ -334,15 +334,22 @@ export class DashboardComponent implements OnInit, AfterViewInit, OnDestroy {
 
     this.core.register({observerClass: this, eventName: 'PoolData'}).subscribe((evt:CoreEvent) => {
       this.pools = evt.data;
-      this.isDataReady();
-    });
 
-    this.core.register({observerClass: this, eventName: 'VolumeData'}).subscribe((evt:CoreEvent) => {
-      const nonBootPools = evt.data.filter(v => v.id !== 'boot-pool');
-      const clone = Object.assign({}, evt);
-      clone.data = nonBootPools;
-      this.setVolumeData(clone);
-      this.isDataReady();
+      if(this.pools.length > 0){
+        this.ws.call('pool.dataset.query', [[], {"extra": {"retrieve_children": false}}] ).subscribe((res) => {
+          const data = res.filter( pool => pool.id !== 'boot-pool');
```

Comment:
I think we should be using `boot.pool_name` to retrieve boot pool name and then we can have this as a filter above when we query for datasets.

Reply:
Yes, boot.pool_name should be called only once and then cached for eternity.